vertical-pod-autoscaler/recommender/model/types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"time"
+)
+
+// MeasuredResourceName represent the name of the resource monitored by recommender
+type MeasuredResourceName string
+
+const (
+	// ResourceCPU represents CPU in milicores (1core = 1000milicores)
+	ResourceCPU MeasuredResourceName = "cpu"
+	// ResourceMemory represents memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+	ResourceMemory MeasuredResourceName = "memory"
+)
+
+// PodID contains information needed to identify a Pod within a cluster
+type PodID struct {
+	// Namespaces where the Pod is defined
+	Namespace string
+	// PodName is a name of the pod unique within a namespace
+	PodName string
+}
+
+// ContainerID contains information needed to identify a Container within a cluster
+type ContainerID struct {
+	PodID
+	// ContainerName is the name of the container, unique within a pod.
+	ContainerName string
+}
+
+// ContainerMetricsSnapshot contains information about usage of certain container withing defined time window
+type ContainerMetricsSnapshot struct { // nazwa zostaje
+	// ID identifies a specific container those metrics are comming from
+	ID ContainerID
+	// Time when the snapshot was taken
+	SnapshotTime time.Time
+	// duration, which this snapshot represents
+	SnapshotWindow time.Duration
+	// Actual usage of resources, during the SnapshotWindow.
+	Usage map[MeasuredResourceName]int
+}
+
+// BasicPodSpec contains basic information defining a pod and its containers
+type BasicPodSpec struct {
+	// ID identifies a pod within a cluster
+	ID PodID
+	// Labels of the pod. It is used to match pods with certain VPA opjects.
+	PodLabels map[string]string
```

Comment:
How about using labels.Set ("k8s.io/apimachinery/pkg/labels")?

Reply:
We already get map[string]string from the Lister API (see [here](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L209)). Re-packaging this map would only add an additional dependency to an external package.